angular/projects/spark-angular/src/lib/components/sprk-tabs/sprk-tabs.component.spec.ts

```diff
@@ -0,0 +1,474 @@
+}
+
+describe('SprkTabsComponent', () => {
+  let component: SprkTabsComponent;
+  let fixture: ComponentFixture<SprkTabsComponent>;
+  let element: HTMLElement;
+
+  let dynamicTestComponent;
+  let dynamicTestFixture;
+
+  let testComponent: TestComponent;
+  let testFixture: ComponentFixture<TestComponent>;
+  let testElement;
+  let testTab1;
+  let testTab2;
+  let testTab3;
+  let testPanel1;
+  let testPanel2;
+  let testPanel3;
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      declarations: [
+        SprkTabsComponent,
+        TestComponent,
+        DynamicComponent,
+        SprkTabsPanelDirective,
+        SprkTabsButtonDirective,
+      ],
+    }).compileComponents();
+  });
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(SprkTabsComponent);
+    component = fixture.componentInstance;
+
+    testFixture = TestBed.createComponent(TestComponent);
+    testComponent = testFixture.componentInstance;
+    testFixture.detectChanges();
+
+    element = fixture.nativeElement.querySelector('div');
+    testElement = testFixture.nativeElement.querySelector('sprk-tabs');
+
+    testTab1 = testElement.querySelectorAll('.sprk-c-Tabs__button')[0];
+    testTab2 = testElement.querySelectorAll('.sprk-c-Tabs__button')[1];
+    testTab3 = testElement.querySelectorAll('.sprk-c-Tabs__button')[2];
+    testPanel1 = testElement.querySelectorAll('.sprk-c-Tabs__content')[0];
+    testPanel2 = testElement.querySelectorAll('.sprk-c-Tabs__content')[1];
+    testPanel3 = testElement.querySelectorAll('.sprk-c-Tabs__content')[2];
+
+    // Dynamic Test Component
+    dynamicTestFixture = TestBed.createComponent(DynamicComponent);
+    dynamicTestComponent = dynamicTestFixture.componentInstance;
+  });
+
+  it('should create itself', () => {
+    expect(component).toBeTruthy();
+  });
+
+  it('should add classes if additionalClasses has a value', () => {
+    component.additionalClasses = 'sprk-u-man';
+    fixture.detectChanges();
+    expect(element.classList.toString()).toEqual('sprk-c-Tabs sprk-u-man');
+  });
+
+  it('should add tabstrip additional classes if tablistAdditionalClasses has a value', () => {
+    component.tablistAdditionalClasses = 'sprk-u-man';
+    fixture.detectChanges();
+    expect(element.querySelector('.sprk-c-Tabs__buttons').classList.toString()).toEqual('sprk-c-Tabs__buttons sprk-u-man');
+  });
+
+  it('should create an id for each tab', () => {
+    expect(testTab1.id).toContain('tabs');
+    expect(testTab2.id).toContain('tabs');
+    expect(testTab1.id).not.toEqual(testTab2.id);
+  });
+
+  it('should create an id for each panel', () => {
+    expect(testPanel1.id).toContain('tabs');
+    expect(testPanel2.id).toContain('tabs');
+    expect(testPanel1.id).not.toEqual(testPanel2.id);
+  });
+
+  it('should set aria-controls on each tab', () => {
+    expect(testTab1.getAttribute('aria-controls')).toEqual(testPanel1.id);
+    expect(testTab2.getAttribute('aria-controls')).toEqual(testPanel2.id);
```

Comment:
This says on each tab but we are only checking tab1 and tab2 here. Missing that 3rd check.